miden-tx/src/executor/mod.rs

```diff
@@ -171,18 +172,20 @@ impl<D: DataStore> TransactionExecutor<D> {
     /// - If required data can not be fetched from the [DataStore].
     /// - If the transaction program can not be compiled.
     /// - If the transaction program can not be executed.
-    pub fn execute_transaction(
+    pub fn execute_transaction<A: TransactionAuthenticator>(
         &self,
         account_id: AccountId,
         block_ref: u32,
         notes: &[NoteId],
         tx_args: TransactionArgs,
+        tx_authenticator: A,
```

Comment:
When thinking of how this would come into play in the client, it seemed to me that authentication would be more a function of the transaction rather than anything else. Since the client instantiates the executor on `Client::new()`, it seemed better here but we can move it if it makes more sense.

Reply:
I was thinking `TransactionAuthenticator` would be defined for the client at instantiation time (similar to how `Store` is defined), and so, we's treat the authenticator the same way as we treat the store component. I guess to figure out how to make it work, we first should finalize https://github.com/0xPolygonMiden/miden-base/pull/609.

Reply:
I pushed these changes and it works, but there are a couple of points worth mentioning:

- I defined the interface signature as `get_signature(&mut self, ...)` due to the base authenticator having to possibly modify the `Rng` (the Falcon signature did this). What this means is that the mutability requirement bubbles up to `execute_transaction`. Also, `TransactionHost` now has a lifetime mutable reference which it gets from the executor.
- Should we provide a `pub fn authenticator() -> &mut Authenticator` getter for the `TransactionExecutor`? From the client then we could instantiate the executor and then add/remove keys as needed before running transactions.

Reply:
Regarding the first point, I left a comment in the review below, but basically, I'm wondering if we can avoid mutable reference to `self` buy using interior mutability within a specific implementation of the `TransactionExecutor` when needed.

Regarding the second point, I'm not sure. I think this would imply that we need additional methods on the authenticator interface - e.g., something like `Authenticator::add_key()` and `Authenticator::remove_key()` - but I'm a bit hesitant to add them now. Either way though, adding `pub fn authenticator() -> &mut Authenticator` in the future should be a non-breaking change - and so, we should be able to do it relatively easily.